src/scripts/main.js

```diff
@@ -1,3 +1,171 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const headers = document.querySelectorAll('th');
+  const tbody = document.querySelector('tbody');
+  let rows = Array.from(tbody.querySelectorAll('tr'));
+  const sortOrder = Array(headers.length).fill(true);
+
+  function updateRows() {
+    rows = Array.from(tbody.querySelectorAll('tr'));
+  }
+
+  function sortColumn(index) {
+    updateRows();
+
+    rows.sort((a, b) => {
+      const cellA = a.cells[index].textContent.trim();
+      const cellB = b.cells[index].textContent.trim();
+      const isCurrency = cellA.includes('$');
+
+      if (isCurrency) {
+        const valA = parseFloat(cellA.replace(/[$,]/g, ''));
+        const valB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+        return sortOrder[index] ? valA - valB : valB - valA;
+      }
+
+      return sortOrder[index]
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    });
+
+    tbody.innerHTML = '';
+    rows.forEach((row) => tbody.appendChild(row));
+    sortOrder[index] = !sortOrder[index];
+  }
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => sortColumn(index));
+  });
+
+  function toggleActiveRow(row) {
+    document
+      .querySelectorAll('tr.active')
+      .forEach((r) => r.classList.remove('active'));
+    row.classList.add('active');
+  }
+
+  tbody.addEventListener('click', (e) => {
+    const row = e.target.closest('tr');
+
+    if (row) {
+      toggleActiveRow(row);
+    }
+  });
+
+  const form = document.createElement('form');
+
+  form.className = 'new-employee-form';
+
+  form.innerHTML = `
+    <label>
+      Name: <input name="name" type="text" data-qa="name" required>
+    </label>
+    <label>
+      Position: <input name="position" type="text" data-qa="position" required>
+    </label>
+    <label>
+      Office:
+      <select name="office" data-qa="office" required>
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age: <input name="age" type="number" data-qa="age" required>
+    </label>
+    <label>
+      Salary: <input name="salary" type="number" data-qa="salary" required>
```

Comment:
Consider adding validation to ensure the salary input is a valid number before parsing it. This will prevent potential errors if the input is not a number.